### ISSUE TITLE
Turning TTL on doesn't cause existing nodes to expire and be removed

### ISSUE THREAD

**Issue by tzneal** [aws/karpenter#3561]
### Version

__Karpenter Version:__ v0.25.0

__Kubernetes Version:__ v1.23.0

### Expected Behavior

Turning on TTL should cause pre-existing nodes that exist to be removed.

### Actual Behavior

Nodes that were already past the TTL when it was enabled weren't removed

### Steps to Reproduce the Problem

Launch nodes and let them get old
Turn on TTL for the node so that the nodes are already expired

### Resource Specs and Logs

N/A

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by njtran** [comment]
Do they never expire? Or does it just take considerably longer than expected?

**Comment by tzneal** [comment]
They didn't expire immediately, not sure how long they waited to see.

**Comment by njtran** [comment]
I just tried this on a v0.27.0 installation and found nodes were expired almost immediately after adding a TTL of 100 seconds, the nodes had been up for 24h.

**Comment by njtran** [comment]
Tried the same for v0.25.0 with an expiration added after the node came online for 5 minutes, added a TTL of 30 seconds, and saw it similarly immediately expired.

**Comment by jonathan-innis** [comment]
Is it possible there's a timezone discrepency here that's causing the time comparison to cause the nodes to not be expired? Also, would be good to see the Provisioner and Nodes specs that didn't so that we came more easily repro

**Comment by liorfranko** [comment]
Hi,

I'm testing it on 0.27.
3 hours a go I added a `ttlSecondsUntilExpired: 86400`.
I'm watching the events and filtering by node `ip-10-207-99-167.ec2.internal` which is up for 8 days:
```
kubectl get events -w | grep ip-10-207-99-167.ec2.internal
5m55s       Warning   FailedInflightCheck              node/ip-10-207-99-167.ec2.internal    Expected 100G of resource ephemeral-storage, but found 20959212Ki (21.5% of expected)
0s          Warning   FailedInflightCheck              node/ip-10-207-99-167.ec2.internal    Expected 100G of resource ephemeral-storage, but found 20959212Ki (21.5% of expected)

```

```
kubectl describe nodes ip-10-207-99-167.ec2.internal
Name:               ip-10-207-99-167.ec2.internal
Roles:              <none>
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/instance-type=c6in.24xlarge
                    beta.kubernetes.io/os=linux
                    failure-domain.beta.kubernetes.io/region=us-east-1
                    failure-domain.beta.kubernetes.io/zone=us-east-1d
                    intent=apps
                    k8s.io/cloud-provider-aws=7e43c8d478ffdd4cb3168691ff335629
                    karpenter.k8s.aws/instance-ami-id=ami-01ced323515f177b0
                    karpenter.k8s.aws/instance-category=c
                    karpenter.k8s.aws/instance-cpu=96
                    karpenter.k8s.aws/instance-encryption-in-transit-supported=true
                    karpenter.k8s.aws/instance-family=c6in
                    karpenter.k8s.aws/instance-generation=6
                    karpenter.k8s.aws/instance-hypervisor=nitro
                    karpenter.k8s.aws/instance-memory=196608
                    karpenter.k8s.aws/instance-pods=737
                    karpenter.k8s.aws/instance-size=24xlarge
                    karpenter.sh/capacity-type=spot
                    karpenter.sh/initialized=true
                    karpenter.sh/provisioner-name=ds-infra-network-online-spot
                    kubernetes.io/arch=amd64
                    kubernetes.io/hostname=ip-10-207-99-167.ec2.internalec2ssa.info
                    kubernetes.io/os=linux
                    node.kubernetes.io/instance-type=c6in.24xlarge
                    nodegroup-name=ds-infra-network-online-spot
                    project=mobile-ds-infra-network-online
                    topology.ebs.csi.aws.com/zone=us-east-1d
                    topology.kubernetes.io/region=us-east-1
                    topology.kubernetes.io/zone=us-east-1d
Annotations:        alpha.kubernetes.io/provided-node-ip: 10.207.99.167
                    csi.volume.kubernetes.io/nodeid: {"ebs.csi.aws.com":"i-0115d92e86393b8ad","secrets-store.csi.k8s.io":"ip-10-207-99-167.ec2.internal"}
                    management.cattle.io/pod-limits: {"cpu":"9","memory":"173436Mi"}
                    management.cattle.io/pod-requests: {"cpu":"91705m","memory":"166580Mi","pods":"13"}
                    node.alpha.kubernetes.io/ttl: 60
                    volumes.kubernetes.io/controller-managed-attach-detach: true
CreationTimestamp:  Tue, 28 Feb 2023 19:54:37 +0200
Taints:             <none>
Unschedulable:      false
Lease:
  HolderIdentity:  ip-10-207-99-167.ec2.internal
  AcquireTime:     <unset>
  RenewTime:       Thu, 09 Mar 2023 12:07:32 +0200
Conditions:
  Type             Status  LastHeartbeatTime                 LastTransitionTime                Reason                       Message
  ----             ------  -----------------                 ------------------                ------                       -------
  Ready            True    Thu, 09 Mar 2023 12:05:54 +0200   Tue, 28 Feb 2023 19:55:12 +0200   KubeletReady                 kubelet is posting ready status
  MemoryPressure   False   Thu, 09 Mar 2023 12:05:54 +0200   Tue, 28 Feb 2023 19:54:53 +0200   KubeletHasSufficientMemory   kubelet has sufficient memory available
  DiskPressure     False   Thu, 09 Mar 2023 12:05:54 +0200   Tue, 28 Feb 2023 19:54:53 +0200   KubeletHasNoDiskPressure     kubelet has no disk pressure
  PIDPressure      False   Thu, 09 Mar 2023 12:05:54 +0200   Tue, 28 Feb 2023 19:54:53 +0200   KubeletHasSufficientPID      kubelet has sufficient PID available
Addresses:
  InternalIP:   10.207.99.167
  ExternalIP:   54.209.160.201
  Hostname:     ip-10-207-99-167.ec2.internal
  InternalDNS:  ip-10-207-99-167.ec2.internal
  InternalDNS:  ip-10-207-99-167.ec2ssa.info
  ExternalDNS:  ec2-54-209-160-201.compute-1.amazonaws.com
Capacity:
  attachable-volumes-aws-ebs:  39
  cpu:                         96
  ephemeral-storage:           20959212Ki
  hugepages-1Gi:               0
  hugepages-2Mi:               0
  memory:                      194922424Ki
  pods:                        737
Allocatable:
  attachable-volumes-aws-ebs:  39
  cpu:                         95690m
  ephemeral-storage:           18242267924
  hugepages-1Gi:               0
  hugepages-2Mi:               0
  memory:                      186257336Ki
  pods:                        737
System Info:
  Machine ID:                 ec2a9cbdb39782aa554a328a5bc5b8b2
  System UUID:                ec2a9cbd-b397-82aa-554a-328a5bc5b8b2
  Boot ID:                    3b1a65a4-9ec7-4498-a7e7-e78195840db8
  Kernel Version:             5.10.165-143.735.amzn2.x86_64
  OS Image:                   Amazon Linux 2
  Operating System:           linux
  Architecture:               amd64
  Container Runtime Version:  containerd://1.6.6
  Kubelet Version:            v1.24.10-eks-48e63af
  Kube-Proxy Version:         v1.24.10-eks-48e63af
ProviderID:                   aws:///us-east-1d/i-0115d92e86393b8ad
Non-terminated Pods:          (13 in total)
  Namespace                   Name                                                             CPU Requests  CPU Limits  Memory Requests  Memory Limits  Age
  ---------                   ----                                                             ------------  ----------  ---------------  -------------  ---
  devops                      csi-secrets-store-provider-aws-jf64x                             50m (0%)      0 (0%)      100Mi (0%)       100Mi (0%)     6d13h
  devops                      csi-secrets-store-provider-aws-secrets-store-csi-driver-d67x5    70m (0%)      0 (0%)      140Mi (0%)       400Mi (0%)     7d21h
  ds-infra-apps               is-ds-online-578775ddbf-wggmq                                    45500m (47%)  4 (4%)      82932Mi (45%)    82932Mi (45%)  19h
  ds-infra-apps               rv-ds-online-575cdd6999-nfpw8                                    45500m (47%)  4 (4%)      82420Mi (45%)    82420Mi (45%)  19h
  granulate-agent             granulate-agent-h8249                                            50m (0%)      1 (1%)      256Mi (0%)       2Gi (1%)       8d
  istio-system                istio-cni-node-tswz9                                             10m (0%)      0 (0%)      50Mi (0%)        256Mi (0%)     8d
  kube-system                 aws-node-nbcnp                                                   25m (0%)      0 (0%)      0 (0%)           0 (0%)         8d
  kube-system                 aws-node-termination-handler-pgd7c                               50m (0%)      0 (0%)      128Mi (0%)       128Mi (0%)     8d
  kube-system                 ebs-csi-node-j88cl                                               150m (0%)     0 (0%)      120Mi (0%)       144Mi (0%)     8d
  kube-system                 kube-proxy-xlmgz                                                 100m (0%)     0 (0%)      0 (0%)           0 (0%)         8d
  kubecost                    kubecost-network-costs-75gdz                                     50m (0%)      0 (0%)      256Mi (0%)       4Gi (2%)       4d1h
  logging                     filebeat-4fzs4                                                   100m (0%)     0 (0%)      128Mi (0%)       512Mi (0%)     8d
  monitoring                  kube-prometheus-stack-prometheus-node-exporter-kbh7q             50m (0%)      0 (0%)      50Mi (0%)        400Mi (0%)     8d
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource                    Requests        Limits
  --------                    --------        ------
  cpu                         91705m (95%)    9 (9%)
  memory                      166580Mi (91%)  173436Mi (95%)
  ephemeral-storage           0 (0%)          0 (0%)
  hugepages-1Gi               0 (0%)          0 (0%)
  hugepages-2Mi               0 (0%)          0 (0%)
  attachable-volumes-aws-ebs  0               0
Events:
  Type     Reason               Age                     From       Message
  ----     ------               ----                    ----       -------
  Warning  FailedInflightCheck  5m11s (x21 over 3h25m)  karpenter  Expected 100G of resource ephemeral-storage, but found 20959212Ki (21.5% of expected)
```

In Karpenter's logs, I see only 2 TTL expired log:
```
2023-03-09T08:27:36.834Z	INFO	controller.deprovisioning	triggering termination for expired node after TTL	{"commit": "dc3af1a", "ttl": "24h0m0s", "delay": "239h52m1.834084192s"}
2023-03-09T08:27:36.834Z	INFO	controller.deprovisioning	deprovisioning via expiration replace, terminating 1 nodes ip-10-207-234-242.ec2.internal/c6in.24xlarge/spot and replacing with node from types r6id.16xlarge, r6in.12xlarge, m6idn.12xlarge, c6a.16xlarge, r6idn.16xlarge and 25 other(s)	{"commit": "dc3af1a"}

2023-03-09T09:53:07.068Z	INFO	controller.deprovisioning	triggering termination for expired node after TTL	{"commit": "dc3af1a", "ttl": "24h0m0s", "delay": "241h16m15.068511416s"}
2023-03-09T09:53:07.068Z	INFO	controller.deprovisioning	deprovisioning via expiration replace, terminating 1 nodes ip-10-207-69-136.ec2.internal/c6in.24xlarge/spot and replacing with 2 spot and 2 on-demand nodes from types r6in.12xlarge, m6idn.12xlarge, r6in.24xlarge, c6a.24xlarge, c6in.24xlarge and 27 other(s)	{"commit": "dc3af1a"}
```

While we have many nodes that already expired.

**Comment by njtran** [comment]
Thanks for the specs and logs. One thought I might have is that it looks like some nodes are terminating, and Karpenter will only terminate one at a time. Could you share how long you were watching the nodes to find all the expired logs? Since it looks like this node in particular is a fairly large instance type `c6in.24xlarge`, I'd be interested to know if you are noticing progressively more nodes be expired, but it might just take a while for the deprovisioning to reschedule all pods onto other nodes, or if provisioning a replacement is taking too long. 

Additionally, we will prioritize expiration on nodes that are most past their expiration time. If you notice some nodes progressively terminating, you may be able to predict which one is the next to expire by sorting nodes by creationTimestamp.

**Comment by liorfranko** [comment]
One at a time, but at what interval?
I added the TTL 12 hours ago, and I still have 450 nodes that expired but have not been replaced.
And PDB is not blocking.
This is a very noisy spot group with many interruptions, maybe Karpenter replaces the expired nodes only when it's quiet for a long period of time?

**Comment by liorfranko** [comment]
BTW, I restarted Karpenter and after the restart I do see more nodes that are being replaced due to TTL:
```
2023-03-09T20:49:54.030Z	INFO	controller.deprovisioning	triggering termination for expired node after TTL	{"commit": "dc3af1a", "ttl": "24h0m0s", "delay": "225h33m57.030274838s"}
2023-03-09T20:49:54.030Z	INFO	controller.deprovisioning	deprovisioning via expiration delete, terminating 1 nodes ip-10-207-209-163.ec2.internal/c5.12xlarge/spot	{"commit": "dc3af1a"}
2023-03-09T20:49:54.051Z	INFO	controller.termination	cordoned node	{"commit": "dc3af1a", "node": "ip-10-207-209-163.ec2.internal"}
2023-03-09T20:50:17.527Z	DEBUG	controller.awsnodetemplate	discovered subnets	{"commit": "dc3af1a", "awsnodetemplate": "control-plane-egress", "subnets": ["subnet-01ff56558940bc0ee (us-east-1f)"]}
2023-03-09T20:50:18.075Z	DEBUG	controller.node_state	discovered EC2 instance types zonal offerings for subnets	{"commit": "dc3af1a", "node": "ip-10-207-153-82.ec2.internal", "subnet-selector": "{\"karpenter.sh/cluster/delivery-prod-02\":\"owned\"}"}
2023-03-09T20:50:19.404Z	DEBUG	controller.node_state	discovered EC2 instance types zonal offerings for subnets	{"commit": "dc3af1a", "node": "ip-10-200-188-37.ec2.internal", "subnet-selector": "{\"purpose\":\"egress\",\"vpc\":\"sonic-us\"}"}
2023-03-09T20:50:36.470Z	INFO	controller.termination	deleted node	{"commit": "dc3af1a", "node": "ip-10-207-209-163.ec2.internal"}
2023-03-09T20:50:40.605Z	INFO	controller.deprovisioning	triggering termination for expired node after TTL	{"commit": "dc3af1a", "ttl": "24h0m0s", "delay": "225h33m4.605405078s"}
2023-03-09T20:50:40.605Z	INFO	controller.deprovisioning	deprovisioning via expiration delete, terminating 1 nodes ip-10-207-253-142.ec2.internal/c5.12xlarge/spot	{"commit": "dc3af1a"}
2023-03-09T20:50:40.626Z	INFO	controller.termination	cordoned node	{"commit": "dc3af1a", "node": "ip-10-207-253-142.ec2.internal"}
2023-03-09T20:51:13.040Z	INFO	controller.termination	deleted node	{"commit": "dc3af1a", "node": "ip-10-207-253-142.ec2.internal"}
2023-03-09T20:51:17.109Z	INFO	controller.deprovisioning	triggering termination for expired node after TTL	{"commit": "dc3af1a", "ttl": "24h0m0s", "delay": "224h33m34.109915832s"}
2023-03-09T20:51:17.109Z	INFO	controller.deprovisioning	deprovisioning via expiration delete, terminating 1 nodes ip-10-207-230-48.ec2.internal/c5.12xlarge/spot	{"commit": "dc3af1a"}
2023-03-09T20:51:17.127Z	INFO	controller.termination	cordoned node	{"commit": "dc3af1a", "node": "ip-10-207-230-48.ec2.internal"}
2023-03-09T20:51:55.626Z	INFO	controller.termination	deleted node	{"commit": "dc3af1a", "node": "ip-10-207-230-48.ec2.internal"}
2023-03-09T20:52:04.267Z	INFO	controller.deprovisioning	triggering termination for expired node after TTL	{"commit": "dc3af1a", "ttl": "24h0m0s", "delay": "224h29m48.267775272s"}
2023-03-09T20:52:04.267Z	INFO	controller.deprovisioning	deprovisioning via expiration delete, terminating 1 nodes ip-10-207-252-94.ec2.internal/c5.12xlarge/spot	{"commit": "dc3af1a"}
```

**Comment by njtran** [comment]
> One at a time, but at what interval?

For any node deprovisioning, once the replacement node is Ready and the expired node is drained and deleted, Karpenter will then do the same for the next expired node. Looks like Karpenter is deprovisioning about a node every 30 seconds. 

> This is a very noisy spot group with many interruptions, maybe Karpenter replaces the expired nodes only when it's quiet for a long period of time?

That may be possible, Karpenter checks the in-memory representation of the cluster to see if it matches what the APIServer says. It may be that for a lot of interruptions in a large cluster that this could hit that condition more often than normal. Thoughts @jonathan-innis?

**Comment by njtran** [comment]
Ah nvm on the second comment, v0.27.0 doesn't check for that condition, so it wouldn't apply there.

**Comment by liorfranko** [comment]
Another 5 hours and 390 nodes are still experied but not replaced.
```
ip-10-207-102-199.ec2.internal   Ready                         <none>   3d9h 
ip-10-207-103-218.ec2.internal   Ready                         <none>   3d9h 
ip-10-207-106-57.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-109-127.ec2.internal   Ready                         <none>   3d9h 
ip-10-207-109-99.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-111-213.ec2.internal   Ready                         <none>   3d9h 
ip-10-207-114-96.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-114-98.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-116-191.ec2.internal   Ready                         <none>   3d9h 
ip-10-207-116-193.ec2.internal   Ready                         <none>   3d9h 
ip-10-207-116-217.ec2.internal   Ready                         <none>   3d9h 
ip-10-207-118-224.ec2.internal   Ready                         <none>   3d9h 
ip-10-207-118-242.ec2.internal   Ready                         <none>   3d9h 
ip-10-207-118-25.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-119-13.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-119-215.ec2.internal   Ready                         <none>   3d9h 
ip-10-207-119-76.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-120-122.ec2.internal   Ready                         <none>   3d9h 
ip-10-207-120-206.ec2.internal   Ready                         <none>   3d9h 
ip-10-207-121-121.ec2.internal   Ready                         <none>   3d9h 
ip-10-207-121-36.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-123-201.ec2.internal   Ready                         <none>   3d9h 
ip-10-207-125-128.ec2.internal   Ready                         <none>   3d9h 
ip-10-207-125-173.ec2.internal   Ready                         <none>   3d9h 
ip-10-207-126-71.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-127-64.ec2.internal    Ready                         <none>   3d8h 
ip-10-207-192-114.ec2.internal   Ready                         <none>   3d16h
ip-10-207-192-187.ec2.internal   Ready                         <none>   3d16h
ip-10-207-192-27.ec2.internal    Ready                         <none>   3d16h
ip-10-207-192-67.ec2.internal    Ready                         <none>   3d16h
ip-10-207-193-191.ec2.internal   Ready                         <none>   3d13h
ip-10-207-193-54.ec2.internal    Ready                         <none>   3d16h
ip-10-207-193-63.ec2.internal    Ready                         <none>   3d15h
ip-10-207-193-8.ec2.internal     Ready                         <none>   3d16h
ip-10-207-194-103.ec2.internal   Ready                         <none>   3d15h
ip-10-207-194-125.ec2.internal   Ready                         <none>   3d13h
ip-10-207-194-184.ec2.internal   Ready                         <none>   3d13h
ip-10-207-194-225.ec2.internal   Ready                         <none>   3d14h
ip-10-207-194-86.ec2.internal    Ready                         <none>   3d15h
ip-10-207-195-121.ec2.internal   Ready                         <none>   3d16h
ip-10-207-195-122.ec2.internal   Ready                         <none>   3d16h
ip-10-207-195-4.ec2.internal     Ready                         <none>   3d13h
ip-10-207-195-76.ec2.internal    Ready                         <none>   3d13h
ip-10-207-196-113.ec2.internal   Ready                         <none>   3d13h
ip-10-207-196-202.ec2.internal   Ready                         <none>   3d14h
ip-10-207-196-214.ec2.internal   Ready                         <none>   3d15h
ip-10-207-196-220.ec2.internal   Ready                         <none>   3d16h
ip-10-207-196-234.ec2.internal   Ready                         <none>   3d16h
ip-10-207-196-242.ec2.internal   Ready                         <none>   3d16h
ip-10-207-196-249.ec2.internal   Ready                         <none>   3d15h
ip-10-207-196-68.ec2.internal    Ready                         <none>   3d16h
ip-10-207-196-81.ec2.internal    Ready                         <none>   3d15h
ip-10-207-196-92.ec2.internal    Ready                         <none>   3d15h
ip-10-207-197-130.ec2.internal   Ready                         <none>   3d15h
ip-10-207-197-183.ec2.internal   Ready                         <none>   3d16h
ip-10-207-197-210.ec2.internal   Ready                         <none>   3d11h
ip-10-207-198-115.ec2.internal   Ready                         <none>   3d15h
ip-10-207-198-198.ec2.internal   Ready                         <none>   3d12h
ip-10-207-198-225.ec2.internal   Ready                         <none>   3d16h
ip-10-207-198-234.ec2.internal   Ready                         <none>   3d15h
ip-10-207-198-44.ec2.internal    Ready                         <none>   3d16h
ip-10-207-199-171.ec2.internal   Ready                         <none>   3d11h
ip-10-207-199-242.ec2.internal   Ready                         <none>   3d14h
ip-10-207-199-4.ec2.internal     Ready                         <none>   3d16h
ip-10-207-199-9.ec2.internal     Ready                         <none>   3d14h
ip-10-207-199-90.ec2.internal    Ready                         <none>   3d16h
ip-10-207-200-240.ec2.internal   Ready                         <none>   3d16h
ip-10-207-200-6.ec2.internal     Ready                         <none>   3d15h
ip-10-207-201-117.ec2.internal   Ready                         <none>   3d13h
ip-10-207-201-131.ec2.internal   Ready                         <none>   3d15h
ip-10-207-201-175.ec2.internal   Ready                         <none>   3d16h
ip-10-207-201-219.ec2.internal   Ready                         <none>   3d15h
ip-10-207-201-243.ec2.internal   Ready                         <none>   3d13h
ip-10-207-201-91.ec2.internal    Ready                         <none>   3d13h
ip-10-207-202-145.ec2.internal   Ready                         <none>   3d15h
ip-10-207-202-197.ec2.internal   Ready                         <none>   3d16h
ip-10-207-202-215.ec2.internal   Ready                         <none>   3d16h
ip-10-207-202-224.ec2.internal   Ready                         <none>   3d13h
ip-10-207-202-86.ec2.internal    Ready                         <none>   3d13h
ip-10-207-202-90.ec2.internal    Ready                         <none>   3d13h
ip-10-207-203-106.ec2.internal   Ready                         <none>   3d16h
ip-10-207-203-151.ec2.internal   Ready                         <none>   3d11h
ip-10-207-203-197.ec2.internal   Ready                         <none>   3d15h
ip-10-207-203-201.ec2.internal   Ready                         <none>   3d16h
ip-10-207-203-203.ec2.internal   Ready                         <none>   3d11h
ip-10-207-204-110.ec2.internal   Ready                         <none>   3d10h
ip-10-207-204-159.ec2.internal   Ready                         <none>   3d15h
ip-10-207-204-203.ec2.internal   Ready                         <none>   3d13h
ip-10-207-204-33.ec2.internal    Ready                         <none>   3d15h
ip-10-207-204-38.ec2.internal    Ready                         <none>   3d13h
ip-10-207-204-87.ec2.internal    Ready                         <none>   3d13h
ip-10-207-205-123.ec2.internal   Ready                         <none>   3d15h
ip-10-207-205-124.ec2.internal   Ready                         <none>   3d13h
ip-10-207-205-243.ec2.internal   Ready                         <none>   3d14h
ip-10-207-205-30.ec2.internal    Ready                         <none>   3d16h
ip-10-207-206-138.ec2.internal   Ready                         <none>   3d15h
ip-10-207-206-187.ec2.internal   Ready                         <none>   3d16h
ip-10-207-206-226.ec2.internal   Ready                         <none>   3d16h
ip-10-207-206-241.ec2.internal   Ready                         <none>   3d15h
ip-10-207-207-115.ec2.internal   Ready                         <none>   3d15h
ip-10-207-207-12.ec2.internal    Ready                         <none>   3d11h
ip-10-207-207-148.ec2.internal   Ready                         <none>   3d14h
ip-10-207-207-178.ec2.internal   Ready                         <none>   3d15h
ip-10-207-207-18.ec2.internal    Ready                         <none>   3d15h
ip-10-207-207-84.ec2.internal    Ready                         <none>   3d15h
ip-10-207-207-94.ec2.internal    Ready                         <none>   3d15h
ip-10-207-208-0.ec2.internal     Ready                         <none>   3d16h
ip-10-207-208-116.ec2.internal   Ready                         <none>   3d16h
ip-10-207-208-147.ec2.internal   Ready                         <none>   3d16h
ip-10-207-208-18.ec2.internal    Ready                         <none>   3d13h
ip-10-207-208-183.ec2.internal   Ready                         <none>   3d15h
ip-10-207-208-45.ec2.internal    Ready                         <none>   3d12h
ip-10-207-209-12.ec2.internal    Ready                         <none>   3d16h
ip-10-207-209-143.ec2.internal   Ready                         <none>   3d15h
ip-10-207-209-208.ec2.internal   Ready                         <none>   3d16h
ip-10-207-209-25.ec2.internal    Ready                         <none>   3d16h
ip-10-207-209-75.ec2.internal    Ready                         <none>   3d15h
ip-10-207-210-188.ec2.internal   Ready                         <none>   3d16h
ip-10-207-210-192.ec2.internal   Ready                         <none>   3d16h
ip-10-207-210-31.ec2.internal    Ready                         <none>   3d16h
ip-10-207-210-5.ec2.internal     Ready                         <none>   3d16h
ip-10-207-210-80.ec2.internal    Ready                         <none>   3d14h
ip-10-207-210-98.ec2.internal    Ready                         <none>   3d14h
ip-10-207-211-10.ec2.internal    Ready                         <none>   3d16h
ip-10-207-211-108.ec2.internal   Ready                         <none>   3d11h
ip-10-207-211-143.ec2.internal   Ready                         <none>   3d14h
ip-10-207-211-6.ec2.internal     Ready                         <none>   3d11h
ip-10-207-211-71.ec2.internal    Ready                         <none>   3d14h
ip-10-207-211-84.ec2.internal    Ready                         <none>   3d15h
ip-10-207-212-23.ec2.internal    Ready                         <none>   3d16h
ip-10-207-213-101.ec2.internal   Ready                         <none>   3d15h
ip-10-207-213-120.ec2.internal   Ready                         <none>   3d11h
ip-10-207-213-133.ec2.internal   Ready                         <none>   3d14h
ip-10-207-213-135.ec2.internal   Ready                         <none>   3d14h
ip-10-207-213-212.ec2.internal   Ready                         <none>   3d16h
ip-10-207-213-68.ec2.internal    Ready                         <none>   3d13h
ip-10-207-214-227.ec2.internal   Ready                         <none>   3d13h
ip-10-207-214-25.ec2.internal    Ready                         <none>   3d8h 
ip-10-207-214-38.ec2.internal    Ready                         <none>   3d12h
ip-10-207-214-84.ec2.internal    Ready                         <none>   3d14h
ip-10-207-214-94.ec2.internal    Ready                         <none>   3d15h
ip-10-207-215-0.ec2.internal     Ready                         <none>   3d16h
ip-10-207-215-185.ec2.internal   Ready                         <none>   3d15h
ip-10-207-215-210.ec2.internal   Ready                         <none>   3d13h
ip-10-207-215-22.ec2.internal    Ready                         <none>   3d15h
ip-10-207-215-234.ec2.internal   Ready                         <none>   3d12h
ip-10-207-215-27.ec2.internal    Ready                         <none>   3d15h
ip-10-207-215-44.ec2.internal    Ready                         <none>   3d15h
ip-10-207-215-84.ec2.internal    Ready                         <none>   3d15h
ip-10-207-216-121.ec2.internal   Ready                         <none>   3d15h
ip-10-207-216-160.ec2.internal   Ready                         <none>   3d14h
ip-10-207-216-168.ec2.internal   Ready                         <none>   3d15h
ip-10-207-216-213.ec2.internal   Ready                         <none>   3d16h
ip-10-207-216-43.ec2.internal    Ready                         <none>   3d15h
ip-10-207-216-91.ec2.internal    Ready                         <none>   3d13h
ip-10-207-217-159.ec2.internal   Ready                         <none>   3d13h
ip-10-207-217-245.ec2.internal   Ready                         <none>   3d11h
ip-10-207-218-0.ec2.internal     Ready                         <none>   3d14h
ip-10-207-218-106.ec2.internal   Ready                         <none>   3d13h
ip-10-207-218-11.ec2.internal    Ready                         <none>   3d16h
ip-10-207-218-116.ec2.internal   Ready                         <none>   3d15h
ip-10-207-218-174.ec2.internal   Ready                         <none>   3d15h
ip-10-207-218-30.ec2.internal    Ready                         <none>   3d15h
ip-10-207-218-70.ec2.internal    Ready                         <none>   3d16h
ip-10-207-219-135.ec2.internal   Ready                         <none>   3d11h
ip-10-207-219-164.ec2.internal   Ready                         <none>   3d16h
ip-10-207-219-210.ec2.internal   Ready                         <none>   3d15h
ip-10-207-219-242.ec2.internal   Ready                         <none>   3d15h
ip-10-207-219-255.ec2.internal   Ready                         <none>   3d11h
ip-10-207-219-46.ec2.internal    Ready                         <none>   3d16h
ip-10-207-219-89.ec2.internal    Ready                         <none>   3d15h
ip-10-207-220-119.ec2.internal   Ready                         <none>   3d16h
ip-10-207-220-137.ec2.internal   Ready                         <none>   3d11h
ip-10-207-220-154.ec2.internal   Ready                         <none>   3d14h
ip-10-207-220-17.ec2.internal    Ready                         <none>   3d16h
ip-10-207-220-182.ec2.internal   Ready                         <none>   3d15h
ip-10-207-220-20.ec2.internal    Ready                         <none>   3d16h
ip-10-207-220-212.ec2.internal   Ready                         <none>   3d16h
ip-10-207-221-124.ec2.internal   Ready                         <none>   3d16h
ip-10-207-221-172.ec2.internal   Ready                         <none>   3d14h
ip-10-207-221-232.ec2.internal   Ready                         <none>   3d15h
ip-10-207-221-72.ec2.internal    Ready                         <none>   3d16h
ip-10-207-221-78.ec2.internal    Ready                         <none>   3d15h
ip-10-207-222-101.ec2.internal   Ready                         <none>   3d15h
ip-10-207-222-110.ec2.internal   Ready                         <none>   3d16h
ip-10-207-222-187.ec2.internal   Ready                         <none>   3d16h
ip-10-207-222-245.ec2.internal   Ready                         <none>   3d16h
ip-10-207-223-130.ec2.internal   Ready                         <none>   3d13h
ip-10-207-223-217.ec2.internal   Ready                         <none>   3d15h
ip-10-207-223-231.ec2.internal   Ready                         <none>   3d16h
ip-10-207-223-64.ec2.internal    Ready                         <none>   3d11h
ip-10-207-223-95.ec2.internal    Ready                         <none>   3d16h
ip-10-207-224-162.ec2.internal   Ready                         <none>   3d16h
ip-10-207-224-207.ec2.internal   Ready                         <none>   3d16h
ip-10-207-224-247.ec2.internal   Ready                         <none>   3d15h
ip-10-207-224-68.ec2.internal    Ready                         <none>   3d15h
ip-10-207-224-77.ec2.internal    Ready                         <none>   3d13h
ip-10-207-224-90.ec2.internal    Ready                         <none>   3d16h
ip-10-207-225-103.ec2.internal   Ready                         <none>   3d16h
ip-10-207-225-133.ec2.internal   Ready                         <none>   3d13h
ip-10-207-225-181.ec2.internal   Ready                         <none>   3d15h
ip-10-207-225-192.ec2.internal   Ready                         <none>   3d15h
ip-10-207-225-215.ec2.internal   Ready                         <none>   3d16h
ip-10-207-225-229.ec2.internal   Ready                         <none>   3d16h
ip-10-207-225-232.ec2.internal   Ready                         <none>   3d13h
ip-10-207-225-41.ec2.internal    Ready                         <none>   3d13h
ip-10-207-226-64.ec2.internal    Ready                         <none>   3d16h
ip-10-207-227-167.ec2.internal   Ready                         <none>   3d16h
ip-10-207-227-204.ec2.internal   Ready                         <none>   3d15h
ip-10-207-227-225.ec2.internal   Ready                         <none>   3d11h
ip-10-207-227-35.ec2.internal    Ready                         <none>   3d15h
ip-10-207-227-54.ec2.internal    Ready                         <none>   3d15h
ip-10-207-227-73.ec2.internal    Ready                         <none>   3d13h
ip-10-207-228-110.ec2.internal   Ready                         <none>   3d15h
ip-10-207-228-175.ec2.internal   Ready                         <none>   3d12h
ip-10-207-228-178.ec2.internal   Ready                         <none>   3d14h
ip-10-207-228-191.ec2.internal   Ready                         <none>   3d14h
ip-10-207-229-0.ec2.internal     Ready                         <none>   3d16h
ip-10-207-229-140.ec2.internal   Ready                         <none>   3d14h
ip-10-207-229-198.ec2.internal   Ready                         <none>   3d14h
ip-10-207-229-240.ec2.internal   Ready                         <none>   3d16h
ip-10-207-230-107.ec2.internal   Ready                         <none>   3d15h
ip-10-207-230-253.ec2.internal   Ready                         <none>   3d16h
ip-10-207-230-82.ec2.internal    Ready                         <none>   3d13h
ip-10-207-231-126.ec2.internal   Ready                         <none>   3d15h
ip-10-207-231-186.ec2.internal   Ready                         <none>   3d15h
ip-10-207-231-203.ec2.internal   Ready                         <none>   3d15h
ip-10-207-231-246.ec2.internal   Ready                         <none>   3d15h
ip-10-207-231-35.ec2.internal    Ready                         <none>   3d16h
ip-10-207-231-38.ec2.internal    Ready                         <none>   3d13h
ip-10-207-231-81.ec2.internal    Ready                         <none>   3d16h
ip-10-207-232-113.ec2.internal   Ready                         <none>   3d16h
ip-10-207-232-115.ec2.internal   Ready                         <none>   3d15h
ip-10-207-232-173.ec2.internal   Ready                         <none>   3d15h
ip-10-207-232-38.ec2.internal    Ready                         <none>   3d12h
ip-10-207-233-133.ec2.internal   Ready                         <none>   3d15h
ip-10-207-233-152.ec2.internal   Ready                         <none>   3d13h
ip-10-207-233-159.ec2.internal   Ready                         <none>   3d15h
ip-10-207-233-200.ec2.internal   Ready                         <none>   3d16h
ip-10-207-233-254.ec2.internal   Ready                         <none>   3d11h
ip-10-207-233-83.ec2.internal    Ready                         <none>   3d13h
ip-10-207-233-90.ec2.internal    Ready                         <none>   3d16h
ip-10-207-234-119.ec2.internal   Ready                         <none>   3d13h
ip-10-207-234-172.ec2.internal   Ready                         <none>   3d11h
ip-10-207-234-174.ec2.internal   Ready                         <none>   3d11h
ip-10-207-234-217.ec2.internal   Ready                         <none>   3d14h
ip-10-207-234-76.ec2.internal    Ready                         <none>   3d16h
ip-10-207-235-205.ec2.internal   Ready                         <none>   3d16h
ip-10-207-235-212.ec2.internal   Ready                         <none>   3d15h
ip-10-207-235-25.ec2.internal    Ready                         <none>   3d14h
ip-10-207-235-44.ec2.internal    Ready                         <none>   3d14h
ip-10-207-236-124.ec2.internal   Ready                         <none>   3d15h
ip-10-207-236-195.ec2.internal   Ready                         <none>   3d16h
ip-10-207-236-28.ec2.internal    Ready                         <none>   3d15h
ip-10-207-236-68.ec2.internal    Ready                         <none>   3d13h
ip-10-207-236-82.ec2.internal    Ready                         <none>   3d13h
ip-10-207-237-105.ec2.internal   Ready                         <none>   3d15h
ip-10-207-237-136.ec2.internal   Ready                         <none>   3d11h
ip-10-207-237-142.ec2.internal   Ready                         <none>   3d15h
ip-10-207-237-158.ec2.internal   Ready                         <none>   3d15h
ip-10-207-237-208.ec2.internal   Ready                         <none>   3d15h
ip-10-207-238-133.ec2.internal   Ready                         <none>   3d16h
ip-10-207-238-234.ec2.internal   Ready                         <none>   3d14h
ip-10-207-238-252.ec2.internal   Ready                         <none>   3d16h
ip-10-207-238-59.ec2.internal    Ready                         <none>   3d16h
ip-10-207-238-9.ec2.internal     Ready                         <none>   3d16h
ip-10-207-239-107.ec2.internal   Ready                         <none>   3d15h
ip-10-207-239-129.ec2.internal   Ready                         <none>   3d15h
ip-10-207-239-134.ec2.internal   Ready                         <none>   3d16h
ip-10-207-239-20.ec2.internal    Ready                         <none>   3d15h
ip-10-207-239-32.ec2.internal    Ready                         <none>   3d16h
ip-10-207-239-34.ec2.internal    Ready                         <none>   3d15h
ip-10-207-239-78.ec2.internal    Ready                         <none>   3d14h
ip-10-207-240-100.ec2.internal   Ready                         <none>   3d15h
ip-10-207-240-118.ec2.internal   Ready                         <none>   3d16h
ip-10-207-240-171.ec2.internal   Ready                         <none>   3d13h
ip-10-207-240-188.ec2.internal   Ready                         <none>   3d16h
ip-10-207-240-194.ec2.internal   Ready                         <none>   3d16h
ip-10-207-241-131.ec2.internal   Ready                         <none>   3d12h
ip-10-207-241-192.ec2.internal   Ready                         <none>   3d13h
ip-10-207-241-2.ec2.internal     Ready                         <none>   3d13h
ip-10-207-241-248.ec2.internal   Ready                         <none>   3d13h
ip-10-207-241-80.ec2.internal    Ready                         <none>   3d16h
ip-10-207-241-92.ec2.internal    Ready                         <none>   3d16h
ip-10-207-242-103.ec2.internal   Ready                         <none>   3d15h
ip-10-207-242-130.ec2.internal   Ready                         <none>   3d15h
ip-10-207-242-176.ec2.internal   Ready                         <none>   3d16h
ip-10-207-242-219.ec2.internal   Ready                         <none>   3d16h
ip-10-207-242-92.ec2.internal    Ready                         <none>   3d12h
ip-10-207-242-99.ec2.internal    Ready                         <none>   3d12h
ip-10-207-243-0.ec2.internal     Ready                         <none>   3d11h
ip-10-207-243-155.ec2.internal   Ready                         <none>   3d15h
ip-10-207-243-156.ec2.internal   Ready                         <none>   3d13h
ip-10-207-243-91.ec2.internal    Ready                         <none>   3d13h
ip-10-207-243-94.ec2.internal    Ready                         <none>   3d16h
ip-10-207-244-126.ec2.internal   Ready                         <none>   3d16h
ip-10-207-244-194.ec2.internal   Ready                         <none>   3d14h
ip-10-207-244-202.ec2.internal   Ready                         <none>   3d15h
ip-10-207-244-250.ec2.internal   Ready                         <none>   3d16h
ip-10-207-244-4.ec2.internal     Ready                         <none>   3d15h
ip-10-207-244-41.ec2.internal    Ready                         <none>   3d15h
ip-10-207-244-55.ec2.internal    Ready                         <none>   3d16h
ip-10-207-245-141.ec2.internal   Ready                         <none>   3d11h
ip-10-207-245-194.ec2.internal   Ready                         <none>   3d15h
ip-10-207-245-246.ec2.internal   Ready                         <none>   3d16h
ip-10-207-245-87.ec2.internal    Ready                         <none>   3d15h
ip-10-207-246-0.ec2.internal     Ready                         <none>   3d13h
ip-10-207-246-168.ec2.internal   Ready                         <none>   3d16h
ip-10-207-246-171.ec2.internal   Ready                         <none>   3d16h
ip-10-207-246-21.ec2.internal    Ready                         <none>   3d13h
ip-10-207-246-221.ec2.internal   Ready                         <none>   3d16h
ip-10-207-246-76.ec2.internal    Ready                         <none>   3d16h
ip-10-207-247-10.ec2.internal    Ready                         <none>   3d12h
ip-10-207-247-140.ec2.internal   Ready                         <none>   3d11h
ip-10-207-247-145.ec2.internal   Ready                         <none>   3d15h
ip-10-207-247-162.ec2.internal   Ready                         <none>   3d13h
ip-10-207-247-2.ec2.internal     Ready                         <none>   3d13h
ip-10-207-247-246.ec2.internal   Ready                         <none>   3d15h
ip-10-207-247-45.ec2.internal    Ready                         <none>   3d16h
ip-10-207-247-62.ec2.internal    Ready                         <none>   3d12h
ip-10-207-248-145.ec2.internal   Ready                         <none>   3d11h
ip-10-207-248-203.ec2.internal   Ready                         <none>   3d13h
ip-10-207-248-206.ec2.internal   Ready                         <none>   3d16h
ip-10-207-248-37.ec2.internal    Ready                         <none>   3d15h
ip-10-207-248-6.ec2.internal     Ready                         <none>   3d13h
ip-10-207-248-85.ec2.internal    Ready                         <none>   3d16h
ip-10-207-249-103.ec2.internal   Ready                         <none>   3d14h
ip-10-207-249-204.ec2.internal   Ready                         <none>   3d16h
ip-10-207-249-221.ec2.internal   Ready                         <none>   3d16h
ip-10-207-249-244.ec2.internal   Ready                         <none>   3d16h
ip-10-207-249-65.ec2.internal    Ready                         <none>   3d15h
ip-10-207-250-159.ec2.internal   Ready                         <none>   3d15h
ip-10-207-250-165.ec2.internal   Ready                         <none>   3d15h
ip-10-207-250-225.ec2.internal   Ready                         <none>   3d15h
ip-10-207-250-91.ec2.internal    Ready                         <none>   3d12h
ip-10-207-251-104.ec2.internal   Ready                         <none>   3d14h
ip-10-207-251-180.ec2.internal   Ready                         <none>   3d15h
ip-10-207-251-56.ec2.internal    Ready                         <none>   3d13h
ip-10-207-252-123.ec2.internal   Ready                         <none>   3d13h
ip-10-207-252-73.ec2.internal    Ready                         <none>   3d15h
ip-10-207-252-99.ec2.internal    Ready                         <none>   3d15h
ip-10-207-253-135.ec2.internal   Ready                         <none>   3d16h
ip-10-207-253-176.ec2.internal   Ready                         <none>   3d14h
ip-10-207-253-186.ec2.internal   Ready                         <none>   3d15h
ip-10-207-253-187.ec2.internal   Ready                         <none>   3d13h
ip-10-207-253-19.ec2.internal    Ready                         <none>   3d15h
ip-10-207-253-24.ec2.internal    Ready                         <none>   3d16h
ip-10-207-253-25.ec2.internal    Ready                         <none>   3d16h
ip-10-207-253-251.ec2.internal   Ready                         <none>   3d14h
ip-10-207-253-69.ec2.internal    Ready                         <none>   3d13h
ip-10-207-254-219.ec2.internal   Ready                         <none>   3d15h
ip-10-207-255-163.ec2.internal   Ready                         <none>   3d15h
ip-10-207-255-187.ec2.internal   Ready                         <none>   3d16h
ip-10-207-255-194.ec2.internal   Ready                         <none>   3d16h
ip-10-207-255-237.ec2.internal   Ready                         <none>   3d15h
ip-10-207-255-58.ec2.internal    Ready                         <none>   3d15h
ip-10-207-255-83.ec2.internal    Ready                         <none>   3d13h
ip-10-207-255-86.ec2.internal    Ready                         <none>   3d16h
ip-10-207-34-167.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-34-181.ec2.internal    Ready                         <none>   3d16h
ip-10-207-35-200.ec2.internal    Ready                         <none>   3d16h
ip-10-207-65-144.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-70-170.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-70-203.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-70-85.ec2.internal     Ready                         <none>   3d9h 
ip-10-207-71-123.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-71-96.ec2.internal     Ready                         <none>   3d9h 
ip-10-207-72-24.ec2.internal     Ready                         <none>   3d9h 
ip-10-207-73-200.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-78-220.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-81-118.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-81-254.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-82-72.ec2.internal     Ready                         <none>   3d9h 
ip-10-207-83-162.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-84-69.ec2.internal     Ready                         <none>   3d9h 
ip-10-207-85-24.ec2.internal     Ready                         <none>   3d9h 
ip-10-207-87-170.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-88-130.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-88-227.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-90-135.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-90-155.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-90-159.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-90-215.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-90-222.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-91-19.ec2.internal     Ready                         <none>   3d9h 
ip-10-207-92-241.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-93-161.ec2.internal    Ready                         <none>   3d9h 
ip-10-207-94-60.ec2.internal     Ready                         <none>   3d9h 
ip-10-207-97-181.ec2.internal    Ready                         <none>   3d9h 
```

**Comment by liorfranko** [comment]
We experienced another issue now.
Once a queue of nodes is waiting to be deleted by the expired TTL, consolidation stops working.
We have 274 nodes with no pod that hasn't been deleted from the cluster.

**Comment by njtran** [comment]
Thanks for the logs, if consolidation took place once you removed expiration, i'm fairly confident that it's due to how deprovisioning waits for there to be no expired nodes before computing consolidation actions. 

For reference, if you're able to deprovision a node every 60 seconds, for 2000 nodes, this would take 33.333 hours, which is larger than your 24 hour expiry.

**Comment by njtran** [comment]
Opened up an issue for the parallel deprovisioning above ^
